### PR TITLE
Extension mechanism AWS testing

### PR DESCRIPTION
**Pull request by sblair-metrostar**
initial fixes to aws config validation and cloudflare dns autoprovisioning

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

## Any other comments?

Test plan

- [x] Log into keycloak as root
  - [x] Add a user
- [x] Log into conda-store and create
  - [x] conda environment in a shared namespace
  - [x] ... in personal workspace
- [x] Launch a dask gateway cluster, test autoscaler
  - [x] Validate dask lab extension is working
- [x] Create basic CDS dashboard
- [x] Open VS Code extension
  - [x] Add python extenion
  - [x] Create .py file and run it

### CODE REVIEW

File: src/_nebari/stages/kubernetes_ingress/__init__.py [hunk 1]
```diff
@@ -224,6 +230,17 @@ def set_outputs(
 
         super().set_outputs(stage_outputs, outputs)
 
+    def post_deploy(self, stage_outputs: Dict[str, Dict[str, Any]]):
```
Comment (costrouc) [inline src/_nebari/stages/kubernetes_ingress/__init__.py:233]: This was added due to the auto-provisioning not working. We need to make sure this is preserved.

**Comment by zcfwqg** [comment]
6

File: src/_nebari/render.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def list_files(
             deleted_files.add(path)
 
     for filename in list_files(output_base_dir, ignore_filenames, ignore_directories):
-        relative_path = pathlib.Path(os.path.relpath(filename, output_base_dir))
+        relative_path = str(pathlib.Path(os.path.relpath(filename, output_base_dir)))
```
Comment (sblair-metrostar) [inline src/_nebari/render.py:156]: Fix for a bug related to mixed use of PosixPath and str values in the lists being returned

File: src/_nebari/stages/kubernetes_ingress/__init__.py [hunk 3]
```diff
@@ -75,7 +79,7 @@ def provision_ingress_dns(
         )
 
     if not disable_checks:
-        checks.check_ingress_dns(stage_outputs, config, disable_prompt)
+        check_ingress_dns(stage_outputs, config, disable_prompt)
```
Comment (sblair-metrostar) [inline src/_nebari/stages/kubernetes_ingress/__init__.py:82]: Fix for a compile error, `checks` didn't exist

File: src/_nebari/render.py (hunk 2 above)
Comment (iameskild) [reply src/_nebari/render.py:156]: I resolved this issue upstream using `pathlib.Path.relative_to` so we and remove this.

**Comment by iameskild** [comment]
I opened this [PR](https://github.com/MetroStar/nebari/pull/1) to merge the `extension-mechanism` branch into this one.

**Review by iameskild — APPROVED** [review]
Thanks for this @sblair-metrostar! We will merge this now and then resolve any further issues on the `extension-mechanism` branch.

**Pull request merged**